src/platform/plugins/shared/dashboard/public/dashboard_api/unified_search_manager.ts

```diff
@@ -132,7 +133,13 @@ export function initializeUnifiedSearchManager(
       if (timeslice !== timeslice$.value) timeslice$.next(timeslice);
     })
   );
-
+  controlGroupSubscriptions.add(
+    esqlVariablesService.esqlVariables$.subscribe((variables) => {
+      if (variables.length) {
+        return panelsReload$.next();
```

Comment:
Would it be possible to only reload panels that use variables instead of reloading all panels? Maybe we could create an issue for this so its not lost.

Reply:
I will yes

Reply:
Done https://github.com/elastic/kibana/issues/206558